code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -38,7 +38,7 @@
 /obj/structure/closet/Initialize(mapload)
 	..()
 	if(mapload && !opened)		// if closed, any item at the crate's loc is put in the contents
-		take_contents()
+		addtimer(CALLBACK(src, .proc/take_contents), 0)	//have to wait for the spawners to run

```

Comment:
Yes, because INVOKE_ASYNC will run it immediately. We want to wait until after all atoms are initialized to do this and timers don't actually fire until all SS's are initialized. I could use SSatoms built in initialization but that'd require adding an extra datum var because we need to... One second I just thought of a better way.